lib/delete-users-using-local-file.rb

```diff
@@ -0,0 +1,109 @@
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+require 'date'
+
+require_relative 'zendesk-setup.rb'
+
+# Setup vars
+
+diag = "true"
+url = "#{ENV['ZENDESK_URL']}/deleted_users/"
+lastyear = Date.today.next_day - 365
+
+File.open "data/soft_deleted_users.json" do |input_file|
+
+  file_data = JSON.load input_file
+
+  file_data.each do |user|
+
+
+    # Extract some fields
+    updated_at = user["updated_at"]
+    last_login_at = user["last_login_at"]
+    user_id = user["id"]
+    active = user["active"]
+    name = user["name"]
+
+    # Convert strings to dates, if strings are nil then set a false date of 2012 so comparison works.
+    # Booleans returned by API are not true boolean but strings.
+
+    if active == "false"
+      # account is already soft deleted so force hard delete
+      hard_delete = "true"
+    end
+
+    false_date = "2012-01-01"
+
+    if hard_delete != "true"
+      if updated_at.nil?
+        updated_at = false_date
+      end
+
+      # parse the date so we can do comparisons
+
+      updated = Date.parse(updated_at)
+
+      if last_login_at != nil
+        last_login = Date.parse(last_login_at)
+      else
+        last_login_at = false_date
+        last_login = Date.parse(last_login_at)
+      end
```

Comment:
`last_login` variable is created within if condition. When a program goes out of if condition, the variable is destroyed or lost. Can we replace these lines 47-52 with the following code?

```
      if last_login_at.nil?
        last_login_at = false_date
      end
      last_login = Date.parse(last_login_at)
```

Reply:
Resolved as suggested